chrysom/client.go

```diff
@@ -354,28 +370,56 @@ func (c *Client) Start(ctx context.Context) error {
 		return ErrUndefinedIntervalTicker
 	}
 
+	c.observer.mu.Lock()
```

Comment:
You can certainly do this with a lock.  However, the CAS pattern is way simpler for this.  You don't need separate fields for a mutex and a state.

If you need some sort of control channel for the goroutine, it's pretty simple to just use an `atomic.Value` for that channel, which can also double as the atomic started/stopped state.